### PR TITLE
Add clients page

### PR DESCRIPTION
**Pull request by CasperWA**
The clients page showcases the various available clients to query OPTIMADE databases.

There are currently 3 different clients:
- Swagger/OpenAPI UI (petstore.swagger.io)
- MaterialsCloud Tool (materialscloud.org/optimadeclient)
- optimade.science

To access this new page, a new menu item "Try It!" has been added.

I have requested reviews from the "management", providers I have explicitly mentioned, and client creators, and would especially like all entities who are mentioned on the new page to give their consent.

### CODE REVIEW

**Review by shyamd — APPROVED** [review]

**Review by ml-evs — APPROVED** [review]
:+1:

**Comment by CasperWA** [comment]
@blokhin is this page okay with you? Since it mentions you by name and points to your website, I want to make sure you're okay with it and/or if the text should be changed slightly.

@ml-evs Could I perhaps ask you to test this locally as well, like you did #29? Specifically if the menu does not look weird now with the extra entry.

File: clients.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+**Try this**:
+
+- [Open Database of Crystals (ODBX)](https://petstore.swagger.io/?url=https://optimade.odbx.science/v1/extensions/openapi.json){:target=_blank}
+- [The Materials Project](https://petstore.swagger.io/?url=https://optimade.materialsproject.org/v1/extensions/openapi.json){:target=_blank}
+- [MaterialsCloud (sample DB)](https://petstore.swagger.io/?url=https://aiida-dev.materialscloud.org/optimade-sample/optimade/v1/extensions/openapi.json){:target=_blank}
+
+## MaterialsCloud Tool
+
+An [open-source](https://github.com/CasperWA/voila-optimade-client){target=_blank} web and local executable client developed by [Casper W. Andersen (THEOS, EPFL)](https://casper.welzel.nu){target=_blank} using [Voilà](https://voila.readthedocs.io){:target=_blank} is available on [MaterialsCloud](https://materialscloud.org){:target=_blank}.
```
Comment (ml-evs) [inline clients.md:19]: Some of the link rendering is broken:

```suggestion
An [open-source](https://github.com/CasperWA/voila-optimade-client){:target=_blank} web and local executable client developed by [Casper W. Andersen (THEOS, EPFL)](https://casper.welzel.nu){:target=_blank} using [Voilà](https://voila.readthedocs.io){:target=_blank} is available on [MaterialsCloud](https://materialscloud.org){:target=_blank}.
```

**Review by ml-evs — CHANGES_REQUESTED** [review]
The menu is now a bit wide (wraps onto two lines when given < half the screen width at 1080p), but no big deal. Could potentially remove "Contributors" as a top-level link and put it under the About section and/or move forums to a link under Documentation?

I like the Try It page though (modulo minor fixes below), perhaps the menu button could be a brighter colour (OPTIMADE blue/red?) for emphasis? I guess a redesign is probably on the horizon anyway...

**Comment by blokhin** [comment]
Please, could you delete the _optimade.science_ for now? The current version is malfunctioning, and we are about to roll out the new major update soon (completely rewritten). I will PR it here as soon as it is ready.

**Comment by CasperWA** [comment]
> The menu is now a bit wide (wraps onto two lines when given < half the screen width at 1080p), but no big deal. Could potentially remove "Contributors" as a top-level link and put it under the About section and/or move forums to a link under Documentation?

I can find pros and cons for keeping/removing them:
The Contributors list is quite a long one, not really fitting for a landing page, but it is indeed related to who the consortium is.
For the forum to be used, it needs a special place, hence a menu item makes sense, however, it again indeed goes under the "umbrella" of relating to documentation - so it would make sense to simply highlight on the documentation page. But I would need some input on this idea from @gmrigna, @shyamd and @mkhorton (who I believe set up this forum?).

> (...) perhaps the menu button could be a brighter colour (OPTIMADE blue/red?) for emphasis? I guess a redesign is probably on the horizon anyway...

I was thinking the exact same thing! To catch the eye, but looking at it I would have to change a lot of the CSS and HTML design it seems. Not worth it for now :sweat_smile: 
And I don't know that a redesign in on the horizon as we already just did one...

**Review by ml-evs — APPROVED** [review]
Fair enough, will just approve and let others decide.

I've just added b450a5d which highlights the Try It button in blue, feel free to revert before merging.

**Comment by gmrigna** [comment]
I will leave to @shyamd and @mkhorton to comment about the "Forum" button. I have no particular idea about where to place it...

**Comment by gmrigna** [comment]
The blue button looks good to me...

**Review by gmrigna — APPROVED** [review]

**Comment by ml-evs** [comment]
> @CasperWA could you revert this commit? [optimade.science](https://optimade.science) is back again : )

I've just done this, so we should be good to go!

**Comment by ml-evs** [comment]
@CasperWA could you check that the changes (URLs etc) you made initially are still valid, then merge when you are happy?

**Review by blokhin — APPROVED** [review]

**Comment by blokhin** [comment]
I see the command `{:target=_blank}` is not rendered properly.

**Comment by ml-evs** [comment]
> I see the command `{:target=_blank}` is not rendered properly.

That's fine, it works in the rendered HTML, we don't have to worry about the markdown here

**Comment by ml-evs** [comment]
I've tested all the client links locally, so I'm going to merge this so as not to block #31.

**Comment by gmrigna** [comment]
In the "Swagger" section, the links to the Materials Project and the Materials Cloud lead to "Failed to load API definition."
The problem may be related to cross-origin (CORS) issue ("Access-Control-Allow-Origin").

**Comment by CasperWA** [comment]
> In the "Swagger" section, the links to the Materials Project and the Materials Cloud lead to "Failed to load API definition."
> The problem may be related to cross-origin (CORS) issue ("Access-Control-Allow-Origin").

For Materials Cloud, I think this is related to the issue that one cannot reach the generated OpenAPI JSON for the implementations. See also Materials-Consortia/optimade-python-tools#520, which I think is the cause of the endpoint not being reachable.

**Pull request merged**